### PR TITLE
test(e2e): add e2e tests for upload collection component 

### PR DESCRIPTION
**Pull request by nikvalor**
## Related Issue(s)

closes none

## Description

add e2e tests for core upload collection component

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: e2e/wdio/core/tests/upload-collection.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        fileUploaderInputFile,
+        checkbox,
+        standardButton
+    } = uploadCollectionPage;
+
+    beforeAll(() => {
+        uploadCollectionPage.open();
+    }, 1);
+
+    afterEach(() => {
```
Comment (maxym-valor) [inline e2e/wdio/core/tests/upload-collection.e2e-spec.ts:49]: add pls `WaitForPresent` and `WaitForElDisplayed`

File: e2e/wdio/core/tests/upload-collection.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        refreshPage();
+    }, 2);
+
+    describe('Should check default example', () => {
+        it('should verify basic checks', () => {
```
Comment (maxym-valor) [inline e2e/wdio/core/tests/upload-collection.e2e-spec.ts:54]: you check here 3 different checks, would be better to divide them into 3 iterations to easily debug if something fails. Same for other iterations `verify basic checks`, please

**Review by maxym-valor — APPROVED** [review]

File: e2e/wdio/core/tests/upload-collection.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        standardButton
+    } = uploadCollectionPage;
+
+    beforeAll(() => {
+        uploadCollectionPage.open();
+    }, 1);
+
+    beforeEach(() => {
+        refreshPage();
+        waitForElDisplayed(uploadCollectionPage.title);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/upload-collection.e2e-spec.ts:52]: for better stability on Safari, please change to 
refreshPage():
waitForPresent(uploadCollectionPage.root)
waitForElDisplayed(uploadCollectionPage.title);

File: e2e/wdio/core/pages/upload-collection.po.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    okButton = 'button[style="display: inline-flex;"]';
+    item = '.fd-list__item';
+    emphasizedButton = '.fd-button--emphasized';
+    fileUploaderInputFile = 'input.fd-file-uploader__hidden';
+    checkbox = 'fd-checkbox';
+    standardButton = '.fd-button--standard';
+
+    open(): void {
+        super.open(this.url);
+        waitForElDisplayed(this.root);
```
Comment (Seamoo13) [inline e2e/wdio/core/pages/upload-collection.po.ts:25]: for better stability on Safari, please change to 
super.open(this.url);
waitForPresent(this.root)
waitForElDisplayed(this.title);

File: e2e/wdio/core/tests/upload-collection.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    function checkFileNameChange(example: string): void {
+        scrollIntoView(example);
+        const buttonLength = getElementArrayLength(example + editButton);
+        for (let i = 0; i < buttonLength; i++) {
+            click(example + editButton, i);
+            setValue(example + input, testText);
+            click(okButton);
+            expect(getAlertText()).toBe(acceptAlertText);
+            acceptAlert();
+            expect(getText(example + link, i)).toBe(testText + formatArr[i]);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/upload-collection.e2e-spec.ts:171]: for safari, probably will need to add .trim() at the end of getText method

**Review by Seamoo13 — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Pull request merged**